cstore_fdw.c

```diff
@@ -317,33 +351,33 @@ CStoreProcessUtility(Node * parseTree, const char *queryString,
 
 				if (strncmp(CSTORE_FDW_NAME, objectName, NAMEDATALEN) == 0)
 				{
-					removeCStoreDirectory = true;
+					removeCStoreRelationFiles = true;
 				}
 			}
 
-			CALL_PREVIOUS_UTILITY(parseTree, queryString, context, paramListInfo,
-								  destReceiver, completionTag);
-
-			if (removeCStoreDirectory)
+			if (removeCStoreRelationFiles)
 			{
-				RemoveCStoreDatabaseDirectory(MyDatabaseId);
+				cstoreTableList = CStoreTableList();
 			}
-		}
-		else
-		{
-			ListCell *fileListCell = NULL;
-			List *droppedTables = DroppedCStoreFilenameList((DropStmt *) parseTree);
 
-			CALL_PREVIOUS_UTILITY(parseTree, queryString, context, paramListInfo,
-								  destReceiver, completionTag);
+			CALL_PREVIOUS_UTILITY(parseTree, queryString, context,
+					paramListInfo, destReceiver, completionTag);
 
-			foreach(fileListCell, droppedTables)
+			if (cstoreTableList != NIL)
```

Comment:
on second thought, that line looks extra, removing it since foreach would perform NIL comparison.